lib/testers/test-parser.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * @author Toru Nagashima <https://github.com/mysticatea>
+ */
+"use strict";
+
+const espree = require("espree");
+const Traverser = require("../util/traverser");
+
+/**
+ * Remove `start`/`end` properties from the given node.
+ * @param {ASTNode} node The node to remove.
+ * @returns {void}
+ */
+function removeStartEnd(node) {
+    delete node.start;
+    delete node.end;
```

Comment:
I think it might be confusing for new developers if the `start` and `end` properties are removed. For example, if someone uses `console.log(node)` when running the rule on regular code, it will have the `start` and `end` properties, but the properties will disappear during the test.

Would it work to do this instead?

```js
Object.defineProperty(node, "start", { get() {
    throw new Error("Use node.range[0] instead of node.start");
} };

Object.defineProperty(node, "end", { get() {
    throw new Error("Use node.range[1] instead of node.end");
} };
```